src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java

```diff
@@ -260,6 +277,24 @@ public void periodic() {
                         chassisSpeeds.vyMetersPerSecond = yPID.calculate(getPose().getTranslation().minus(translationTarget).getY());
                 }
 
+                if(balanceTarget != null ) {
+                        if (Math.abs(getGyroscopePitch().getDegrees()) >= AUTO_BALANCE_TOLERANCE_DEGREES) {
+                                chassisSpeeds.vyMetersPerSecond = yPID.calculate(getGyroscopePitch().minus(balanceTarget).getRadians());
+                                totalMetersTranslated += chassisSpeeds.vyMetersPerSecond * (Timer.getFPGATimestamp() - lastUpdateTimeBalanceTarget);
```

Comment:
Could this be based on our existing bot pose model? You could capture the initial pose when the balanceTarget starts and compare with the current pose to see if the bot has translated a certain distance